.github/workflows/build-deps.yml

```diff
@@ -3,20 +3,230 @@ name: Build emsdk dependencies
+    if: ${{ inputs.binaryen_build || inputs.wasm_binaries_build}}
+
+    runs-on: [self-hosted, Windows, ARM64, WASM]
+
+    steps:
+    - name: Build Binaryen for Arm64
+      shell: cmd
+      run: |
+        git clone https://github.com/WebAssembly/binaryen.git --single-branch --depth 1
+        cd binaryen
+        git submodule init
+        git submodule update
+        if exist bin rmdir /s /q bin
+        call "%ProgramFiles%\Microsoft Visual Studio\2022\Preview\VC\Auxiliary\Build\vcvarsall.bat" arm64
+        cmake -G Ninja -DCMAKE_BUILD_TYPE=Release -DCMAKE_CXX_COMPILER_LAUNCHER='ccache' .
+        ninja
+
+
+    - name: Archive binaryen-arm64.zip
+      uses: actions/upload-artifact@v3
+      with:
+        name: binaryen-arm64.zip
+        path: binaryen/binaryen-arm64.zip
+        retention-days: 1
+
+  build-nodejs:
+    if: ${{ inputs.nodejs_build }}
+
+    runs-on: [self-hosted, Windows, ARM64, WASM]
+
+    steps:
+
+    - name: Build Node.js arm64
+      run: |
+        Set-PSDebug -Trace 1
+        git clone https://github.com/Windows-on-ARM-Experiments/node.git --single-branch -b fix-arm64-compilation --depth 1
+        cd node
+        .\vcbuild openssl-no-asm arm64
+
+  build-emscripten:
+    if: ${{ inputs.emscripten_build || inputs.wasm_binaries_build }}
+
+    runs-on: [ubuntu-latest]
+
+    steps:
+
+    - name: Build Emscripten
+      run: |
+        git clone https://github.com/emscripten-core/emscripten.git --single-branch --depth 1
+        cd emscripten
+        ./tools/install.py ../emscripten_package
+        npm ci --production --no-optional ../emscripten_package
+        cd ../emscripten_package
+        zip -r ../emscripten_package.zip .
+
+    - name: Archive emscripten_package.zip
+      uses: actions/upload-artifact@v3
+      with:
+        name: emscripten_package.zip
+        path: emscripten_package.zip
+        retention-days: 1
+
+  build-wasm-binaries:
+    if: ${{ inputs.wasm_binaries_build }}
+    needs: [build-emscripten, build-binaryen, build-llvm-webassembly-arm64]
+
+    runs-on: [windows-latest]
+
+    steps:
+
+    - uses: actions/download-artifact@v3
+      with:
+        name: emscripten_package.zip
+
+    - uses: actions/download-artifact@v3
+      with:
+        name: llvm-arm64.zip
+
+    - uses: actions/download-artifact@v3
+      with:
+        name: binaryen-arm64.zip
+
+    - name: Pack wasm-binaries for Arm64
       run: |
-        set
         ls
+        New-Item -ItemType Directory -Force -Path emsdk/emscripten
+        Expand-Archive -LiteralPath emscripten_package.zip -DestinationPath emsdk\emscripten
+        Expand-Archive -LiteralPath llvm-arm64.zip -DestinationPath emsdk
+        Expand-Archive -LiteralPath binaryen-arm64.zip -DestinationPath emsdk
+        cd emsdk
+        Compress-Archive -Path * -DestinationPath wasm-binaries-arm64
+
+    - name: Archive wasm-binaries-arm64.zip
+      uses: actions/upload-artifact@v3
+      with:
+        name: wasm-binaries-arm64.zip
```

Comment:
1.  Adding `.zip` to the artifact name leads to double zip extension when downloading the artifact from the GitHub list of artifacts.
2.  There is also a double zipping issue when downlading the artifact from GitHub (actions/upload-artifact#39). This issue can be ovecame but uploading unzipped folder as the artifact content and rely on this additional zipping when downloading the artifact.

Reply:
I will try to use without zip extension